src/utils/crypto_utils/src/crypto_lib.c

```diff
@@ -13,6 +13,11 @@
 #include <ctype.h>
 #include <openssl/bn.h>
 #include <openssl/evp.h>
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
It's so ugly... So very, very ugly. 

No recommendation just wanted that in writing.